### PR TITLE
Accessibility updates for docs site

### PR DESCRIPTION
**Pull request by bobwise**
This PR improves support for various types of screen reader navigation techniques on SparkDesignSystem.com. 

[Instructions on how to use these features in macOS VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts#vo-mac-navigation)

### Improved Support for [Landmark Navigation](https://dequeuniversity.com/assets/html/jquery-summit/html5/slides/landmarks.html)

* Reorder left side navigation so that all elements in the section are included in the `<nav>` landmark element.
* Change the upgrade banner from a `div` to an `aside`.

### Improved Support for [Frame Navigation](https://webaim.org/techniques/frames/#title)

* Added `title` to Youtube iframes.
* Added `title` to all component preview iframes.

### Improved Support for [Header Navigation](https://webaim.org/techniques/screenreader/#navigating)

* Changed section headers on Icons page to avoid duplicates

### Improved Support for [Link Navigation](https://webaim.org/techniques/screenreader/#navigating)

* Redid the alt text of the images on the homepage. These had alt text that described the image but did not describe the link. [When an image is the only content of a link, the text alternative for the image describes the unique function of the link.](https://www.w3.org/WAI/WCAG21/Techniques/html/H30)

### Narrow Layout Improvements

* Reordered narrow nav items to match the order of the wide nav items
* Added additional CSS styles to the code samples in the guides so that code samples render without requiring horizontal scrolling [at a width of 320px](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html)

### Color Contrast Improvements

* Switched code samples to a color theme with higher color contrast 

## Resolved Issues
* #3251 
* #3254

## Related Issues (Not solved in this PR)
* #3300 
* #3252 
* #3737
* #3738
* #3739 
* #3740 
* #3741 
* #3742

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -67,22 +67,13 @@ const Header = ({ setContext }) => {
     element: Link,
   }));
 
-  const usingSparkExamples = useUsingSparkData().examples.map((page) => ({
```
Comment (afebbraro) [inline src/components/Header.js:70]: Why'd we remove this?

File: src/components/markdown-render/inlineCode.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import Highlight, { defaultProps } from 'prism-react-renderer';
-import customTheme from 'prism-react-renderer/themes/github';
+import customTheme from 'prism-react-renderer/themes/vsDark';
```
Comment (afebbraro) [inline src/components/markdown-render/inlineCode.js:3]: lets get this theme switch ran by design

File: src/pages/index.js [hunk 3]
```diff
@@ -181,41 +34,241 @@ const IndexPage = () => (
...
-            teaserConfig={teaserFoundations}
-          />
-
-          <SprkCard
-            additionalClasses="docs-c-Card"
-            variant="teaser"
-            teaserConfig={teaserUtils}
-          />
-
-          <SprkCard
```
Comment (afebbraro) [inline src/pages/index.js:214]: Why did you remove the SprkCard?

File: src/pages/using-spark/foundations/typography.mdx [hunk 4]
```diff
@@ -263,6 +272,7 @@ Use for all paragraph and non-heading text.
 <ComponentPreview
   componentName="typography--body-two"
   componentType="foundations"
+  title="Body Two Example"
```
Comment (afebbraro) [inline src/pages/using-spark/foundations/typography.mdx:275]: We should avoid using title as a prop name. I know it doesnt do this for react, but for angular it causes a11y errors. So we'll just want to use something else. ariaTitle, attrTitle, something like that is fine.

File: src/components/Header.js (hunk 1 above)
Comment (bobwise) [reply src/components/Header.js]: I initially removed it because the folder is empty. I took another look and saw that we're pulling this folder and also a Guides folder in the widescreen navigation, so I added both of those to the narrow nav as well. The goal is to have both navs be in the same order.

File: src/components/InstallingSparkDocsMenu.js [hunk 5]
```diff
@@ -7,9 +7,9 @@ const InstallingSparkDocsMenu = () => {
   const pages = useInstallingSparkData();
 
   return (
-    <>
-      <h3 className="docs-menu__heading">Installing Spark</h3>
+    <div className="docs-menu__collection">
       <nav aria-label="Installing Spark">
+        <h3 className="docs-menu__heading">Installing Spark</h3>
```
Comment (RVMendoza) [inline src/components/InstallingSparkDocsMenu.js:12]: love that we're adding semantics!

File: src/components/markdown-render/inlineCode.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import Highlight, { defaultProps } from 'prism-react-renderer';
-import customTheme from 'prism-react-renderer/themes/github';
+import customTheme from 'prism-react-renderer/themes/duotoneDark';
```
Comment (RVMendoza) [inline src/components/markdown-render/inlineCode.js:3]: This change mentions that this creates a *higher* contrast, but was it a necessary change?

Did the github theme not have adequate enough contrast? If it doesn't, then I'm all in for some kind of change.

We chose this theme because it matched our branding the best. Is there a different theme that has good contrast and matches our theme?

File: src/pages/installing-spark/installing-icons.mdx [hunk 7]
```diff
@@ -20,7 +20,7 @@ downloading the latest SVG, from the correct CDN URL,
 as part of your build process. That way, you’ll be
 able to embed the SVG directly into your compiled site.
 
-### Pros:
+### Advantages to Compile Time
```
Comment (RVMendoza) [inline src/pages/installing-spark/installing-icons.mdx:23]: Yes! Love that this is even clearer!

File: src/components/markdown-render/inlineCode.js [hunk 8]
```diff
@@ -10,7 +9,7 @@ const InlineCode = ({ children, className, additionalPreClasses, theme }) => {
       {...defaultProps}
       code={children}
       language={language}
-      theme={theme || customTheme}
+      theme={undefined}
```
Comment (RVMendoza) [inline src/components/markdown-render/inlineCode.js:12]: What happens if we make this undefined?

Comment (bobwise) [reply src/components/markdown-render/inlineCode.js:12]: If this is set to undefined, the `<Highlight>` component will not load a theme from `node_modules` and will instead fallback to just applying CSS classes to the code sample. And then our new CSS will kick in and apply the styles.

Comment (bobwise) [reply src/components/markdown-render/inlineCode.js:12]: I tried deleting the line and it didn't work. It needs to be explicitly set to `undefined` I guess.

Comment (afebbraro) [reply src/components/markdown-render/inlineCode.js:12]: What benefits do we get from still using the Highlight component with this change?

File: src/scss/vendor/_prism.scss [hunk 9]
```diff
@@ -0,0 +1,143 @@
+/* Generated with http://k88hudson.github.io/syntax-highlighting-theme-generator/www */
```
Comment (afebbraro) [inline src/scss/vendor/_prism.scss:1]: @bobwise  did you check on this about MIT?

**Review by afebbraro — CHANGES_REQUESTED** [review]
left a few comments

**Review by guern1kn — APPROVED** [review]

File: src/components/markdown-render/inlineCode.js (hunk 2 above)
Comment (bobwise) [reply src/components/markdown-render/inlineCode.js]: Design has signed off on this.

File: src/pages/index.js (hunk 3 above)
Comment (bobwise) [reply src/pages/index.js]: This has been added back and new issues created to update SprkCard

File: src/pages/using-spark/foundations/typography.mdx (hunk 4 above)
Comment (bobwise) [reply src/pages/using-spark/foundations/typography.mdx]: This has been updated

File: src/components/markdown-render/inlineCode.js (hunk 6 above)
Comment (bobwise) [reply src/components/markdown-render/inlineCode.js]: We ended up switching to a color scheme based on the Spark colors.

File: src/components/markdown-render/inlineCode.js (hunk 8 above)
Comment (bobwise) [reply src/components/markdown-render/inlineCode.js:12]: The Highlight component adds the Prism classnames that we use to theme the code. We still need it even if we're not using React theming.

File: src/scss/vendor/_prism.scss (hunk 9 above)
Comment (bobwise) [reply src/scss/vendor/_prism.scss:1]: @yodasw16 signed off on this

File: src/pages/index.js [hunk 10]
```diff
@@ -226,12 +226,13 @@ const IndexPage = () => (
           </SprkStackItem>
 
           <SprkStackItem>
-            <SprkButton
+            <SprkLink
```
Comment (afebbraro) [inline src/pages/index.js:229]: This can stay a SprkButton. We have a prop here for element="a" for this specific use case.

**Comment by afebbraro** [comment]
Just left one comment on this that I messaged you about!

File: src/pages/index.js (hunk 10 above)
Comment (bobwise) [reply src/pages/index.js:229]: Thanks Amber! I made this change so that the Learn More link would be included in Link Navigation. [When rendering SprkButton as a link](https://react.sparkdesignsystem.com/?path=/story/components-button--as-a-link-element), the output is still tagged as semantically a button with `role="button"`. This causes it to not be included in Link Navigation. I don't think `SprkButton` works for our use case.

Comment (afebbraro) [reply src/pages/index.js:229]: Making a bug issue for this one as we chatted about.

**Review by afebbraro — APPROVED** [review]
WOOT
